client/src/reader/components/AdventureListItem/AdventureListItem.js

```diff
@@ -1,17 +1,77 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
+import { popModal, popToast } from '../../../shared/components/';
+import closeModal from '../../../shared/components/Modal/closeModal';
 import * as routes from '../../constants/routes';
 import { removeFromLibrary } from '../../store/actions/libraryActions';
+import { addReview, updateReview } from '../../store/actions/reviewActions';
+import reviewsSelector from '../../store/selectors/reviewsSelector';
+import ReviewEditor from '../ReviewEditor';
 import styles from './AdventureListItem.module.css';
 
-const AdventureListItem = ({ adventure, removeFromLibrary }) => {
+const AdventureListItem = ({
+  adventure,
+  removeFromLibrary,
+  addReview,
+  updateReview,
+  reviews,
+}) => {
   const handleRemove = () => {
     if (window.confirm('Remove from your library?')) {
       removeFromLibrary(adventure.id);
     }
   };
 
+  const handleAddReviewClick = e => {
```

Comment:
I'm explicitly using "Add Review" to mean that you can only add a review if one does not already exist.